cmd/site-replication.go

```diff
@@ -1689,3 +1691,584 @@ func getMissingSiteNames(oldDeps, newDeps set.StringSet, currSites []madmin.Peer
+		}
+		for group, policy := range sri.GroupPolicies {
+			if _, ok := userPolicyStats[group]; !ok {
+				groupPolicyStats[group] = make([]srGroupPolicyMapping, 0, numSites)
+			}
+			groupPolicyStats[group] = append(groupPolicyStats[group], srGroupPolicyMapping{SRPolicyMapping: policy, DeploymentID: sri.DeploymentID})
+		}
+	}
+	info.StatsSummary = make(map[string]madmin.SRSiteSummary, len(c.state.Peers))
+	info.BucketMismatches = make(map[string]map[string]madmin.SRBucketStatsSummary)
+	info.PolicyMismatches = make(map[string]map[string]madmin.SRPolicyStatsSummary)
+	info.UserMismatches = make(map[string]map[string]madmin.SRUserStatsSummary)
+	info.GroupMismatches = make(map[string]map[string]madmin.SRGroupStatsSummary)
+	// collect user policy mapping replication status across sites
+	for u, pslc := range userPolicyStats {
+		var policySet = set.NewStringSet()
+		uPolicyCount := 0
+		for _, ps := range pslc {
+			policyBytes, err := json.Marshal(ps)
+			if err != nil {
+				continue
+			}
+			uPolicyCount++
+			if policyStr := string(policyBytes); !policySet.Contains(policyStr) {
+				policySet.Add(policyStr)
+			}
+		}
+		policyMismatch := !isReplicated(uPolicyCount, numSites, policySet)
+		if policyMismatch {
+			for _, ps := range pslc {
+				dID := depIdxes[ps.DeploymentID]
+				_, hasUser := sris[dID].UserPolicies[u]
+
+				info.UserMismatches[u][ps.DeploymentID] = madmin.SRUserStatsSummary{
+					PolicyMismatch: policyMismatch,
+					UserMissing:    !hasUser,
+				}
+			}
+		}
+	}
+
+	// collect user policy mapping replication status across sites
+
+	for g, pslc := range groupPolicyStats {
+		var policySet = set.NewStringSet()
+		gPolicyCount := 0
+		for _, ps := range pslc {
+			policyBytes, err := json.Marshal(ps)
+			if err != nil {
+				continue
+			}
+			gPolicyCount++
+			if policyStr := string(policyBytes); !policySet.Contains(policyStr) {
+				policySet.Add(policyStr)
+			}
+		}
+		policyMismatch := !isReplicated(gPolicyCount, numSites, policySet)
+		if policyMismatch {
+			for _, ps := range pslc {
+				dID := depIdxes[ps.DeploymentID]
+				_, hasGroup := sris[dID].GroupPolicies[g]
+
+				info.GroupMismatches[g][ps.DeploymentID] = madmin.SRGroupStatsSummary{
+					PolicyMismatch: policyMismatch,
+					GroupMissing:   !hasGroup,
+				}
+			}
+		}
+	}
+	// collect IAM policy replication status across sites
+
+	for p, pslc := range policyStats {
+		var policies []*iampolicy.Policy
+		uPolicyCount := 0
+		for _, ps := range pslc {
+			plcy, err := iampolicy.ParseConfig(bytes.NewReader(ps.policy))
+			if err != nil {
+				continue
+			}
+			policies = append(policies, plcy)
+			uPolicyCount++
+			sum := info.StatsSummary[ps.DeploymentID]
+			sum.TotalIAMPoliciesCount++
+			info.StatsSummary[ps.DeploymentID] = sum
+		}
+		policyMismatch := !isIAMPolicyReplicated(uPolicyCount, numSites, policies)
+		if policyMismatch {
+		}
+
```

Comment:
is there something missing here? @poornas

Reply:
Removed this